### PR TITLE
feat(forge): blobbasefee cheatcode

### LINKED ISSUE #7002: Add cheatcode to set `block.blobbasefee`
### Component

Forge

### Describe the feature you would like

Add a cheatcode to set `block. blobbasefee`, similar to how we can set other block fields.

This is a new field on `block` in Solidity 0.8.24, only available in Cancun.

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by kamuik16**
Closes #7002

### CODE REVIEW

File: crates/cheatcodes/spec/src/vm.rs [hunk 1]
```diff
@@ -365,6 +365,16 @@ interface Vm {
     #[cheatcode(group = Evm, safety = Safe)]
     function getBlockTimestamp() external view returns (uint256 timestamp);
 
+    /// Sets `block.blobbasefee`
+    #[cheatcode(group = Evm, safety = Unsafe)]
+    function blobBaseFee(uint256 newBlobBaseFee) external;
+
+    /// Gets the current `block.blobbasefee`.
+    /// You should use this instead of `block.blobbasefee` if you use `vm.blobBaseFee`, as `block.blobbasefee` is assumed to be constant across a transaction,
+    /// and as a result will get optimized out by the compiler.
```
Comment (Evalir) [inline crates/cheatcodes/spec/src/vm.rs:374]: let's also add the issue link here

**Review by Evalir — CHANGES_REQUESTED** [review]
needs forge fmt + nit. This needs to be run using the cancun test profile

File: crates/cheatcodes/spec/src/vm.rs (hunk 1 above)
Comment (kamuik16) [reply crates/cheatcodes/spec/src/vm.rs:374]: > let's also add the issue link here

which issue should I link here?

**Comment by kamuik16** [comment]
> Thanks! needs forge fmt + nit. tests need to be run using the cancun test profile

ran the test with cancun evm version but did not push into the pr.

**Comment by yash-atreya** [comment]
Hey @kamuik16, let me know if you need help with anything

**Comment by kamuik16** [comment]
> Hey @kamuik16, let me know if you need help with anything

I am waiting for reviews. I think it's done from my side. Working correctly.

**Review by mattsse — APPROVED** [review]
lgtm

**Comment by yash-atreya** [comment]
> > Hey @kamuik16, let me know if you need help with anything
> 
> I am waiting for reviews. I think it's done from my side. Working correctly.
CI tests are failing

**Review by mattsse — CHANGES_REQUESTED** [review]
ah this is now a cheat that only works with CANCUN, hence the failing test.

so what we need now is a test, just like:

https://github.com/foundry-rs/foundry/blob/master/crates/forge/tests/it/cheats.rs#L10-L26

but uses 

https://github.com/foundry-rs/foundry/blob/master/crates/forge/tests/it/test_helpers.rs#L273-L275

and we need a cancun/cheats folder not default/cheats

## TODO
* create cancun/cheats folder and move test
* add a new rust test similar to cheats that uses the `TEST_DATA_CANCUN` config

**Comment by kamuik16** [comment]
i will implement this by tomorrow, sounds good?

> ah this is now a cheat that only works with CANCUN, hence the failing test.
> 
> so what we need now is a test, just like:
> 
> https://github.com/foundry-rs/foundry/blob/master/crates/forge/tests/it/cheats.rs#L10-L26
> 
> but uses
> 
> https://github.com/foundry-rs/foundry/blob/master/crates/forge/tests/it/test_helpers.rs#L273-L275
> 
> and we need a cancun/cheats folder not default/cheats
> 
> ## TODO
> * create cancun/cheats folder and move test
> * add a new rust test similar to cheats that uses the `TEST_DATA_CANCUN` config

i'll try and get it done by tomorrow eod, sounds good?

**Comment by kamuik16** [comment]
hey @mattsse! I moved the test into cancun folder and ci is passing but struggling with the test.

**Review by mattsse — CHANGES_REQUESTED** [review]
lgtm

**Review by mattsse — APPROVED** [review]

**Comment by kamuik16** [comment]
> lgtm

thanks for the test help!

**Pull request merged**